server/src/websocket/server.rs

```diff
@@ -789,19 +963,22 @@ impl Handler<Disconnect> for ChatServer {
 
 /// Handler for Message message.
 impl Handler<StandardMessage> for ChatServer {
-  type Result = MessageResult<StandardMessage>;
+  type Result = ResponseFuture<Result<String, failure::Error>>;
 
   fn handle(&mut self, msg: StandardMessage, ctx: &mut Context<Self>) -> Self::Result {
-    match self.parse_json_message(msg, ctx) {
-      Ok(m) => {
-        info!("Message Sent: {}", m);
-        MessageResult(m)
-      }
-      Err(e) => {
-        error!("Error during message handling {}", e);
-        MessageResult(e.to_string())
+    let fut = self.parse_json_message(msg, ctx);
+    Box::pin(async move {
+      match fut.await {
```

Comment:
Ah so this is how you deal with futures in the handles. I looked everywhere for this, but couldn't find anything.